auth/authpb/auth.proto

```diff
@@ -16,7 +16,22 @@ message User {
   int64 tombstone = 3;
 }
 
+// Permission is a single entity
+message Permission {
+  bytes key = 1;
+
+  enum Type {
+    READ = 0;
+    WRITE = 1;
+    READWRITE = 2;
+  }
+  Type perm = 2;
+}
+
 // Role is a single entry in the bucket authRoles
 message Role {
-  bytes name = 2;
+  bytes name = 1;
+
+  repeated Permission keyPermission = 2;
```

Comment:
is there a reason that we cannot add a type into Permission? So that we do not need this two list. And I feel key or prefix should be an attr of permission.